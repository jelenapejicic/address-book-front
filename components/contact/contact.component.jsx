import React from 'react';
import './contact.style.scss';
import {DeleteButton} from '../delete-button/delete-button.component';

export const Contact = (props) => {

    const Modal = props.modal;    
    
    const image_url = props.contact.image_url !== '' ? props.contact.image_url : require('../../assets/images/imagenotavailable.png');
    

    const handleRemoveItem = (e) => {

        //console.log(props.contact.id);

        if(window.confirm('Delete the contact?'))
        {
            const requestOptions = {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify({ id:props.contact.id })
            };
            
            fetch('http://localhost:8080/imenik/api/contact/delete.php', requestOptions).then(props.handleState);

        };
    };
   
    
    return (
        
        <div className="contact-container">
            
            <Modal className="edit-button" iconClass="fa fa-edit" contactDetails={props.contact}></Modal>
            
            <DeleteButton onClick={handleRemoveItem} ></DeleteButton>
            
            <img alt="customer" src={image_url} className="customer-image" />
            <p><span> {props.contact.name} </span>
            <span> {props.contact.last_name} </span></p>
            <p> <i className="fa fa-phone"></i> {props.contact.telephone}</p>
            <p> <i className="fa fa-envelope"></i> {props.contact.email}</p>
        </div>
    );
}
