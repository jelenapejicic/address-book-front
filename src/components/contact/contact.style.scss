.contact-container {
    display: flex;
    flex-direction: column;
    background-color: rgb(206, 211, 209);
    border: 1px solid grey;
    border-radius: 5px;
    padding: 25px;
    cursor: pointer;
    -moz-osx-font-smoothing: grayscale;
    backface-visibility: hidden;
   /* transform: translateZ(0);*/
    /*transition: transform 0.25s ease-out;*/
  }
  
  .customer-image{
    width: 80px;
    margin: auto;
  }
  .delete-icon {
    text-align: right;
  }
  .edit-icon{
    text-align: right;
  }