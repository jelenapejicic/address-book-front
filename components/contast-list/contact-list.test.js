import React from 'react';
import * as enzyme from 'enzyme';
import { ContactList } from './contact-list.component';
import { Contact } from '../contact/contact.component';

describe('ContactList component', () => {
    
    let wrapper;
    const mockList = [
        {
            "id": "1",
            "name": "Jelena",
            "last_name": "Pejicic",
            "email": "jelenapejicic@gmail.com",
            "image_url": "https:\/\/i.pinimg.com\/280x280_RS\/25\/20\/ae\/2520ae445b23f319fbbd5bb254f547fb.jpg",
            "telephone": "011222444"
        },
        {
            "id": "2",
            "name": "Bojana",
            "last_name": "Pejicic",
            "email": "pejicicbojana@gmail.com",
            "image_url": "https:\/\/image.shutterstock.com\/image-photo\/chairs-umbrella-palm-beach-tropical-260nw-559599520.jpg",
            "telephone": "011222333"
        },
    ];

    beforeEach(() => {
       
        const mockProps = {
            contacts: mockList,
            handleState: jest.fn()
        };
    
        wrapper = enzyme.shallow(<ContactList {...mockProps} />);
      
    });

    it('it should render List of Contacts', () => {

        expect(wrapper).toMatchSnapshot();

    });

    it('should render an equal number of Contact components as the ContactList prop', () => {
        
        console.log(wrapper.debug());
        const children = wrapper.find(Contact);
        expect (children).toHaveLength(2);

    });
});
