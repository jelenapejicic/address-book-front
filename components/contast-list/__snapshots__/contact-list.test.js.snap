// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList component it should render List of Contacts 1`] = `ShallowWrapper {}`;

exports[`it should render List of Contacts 1`] = `ShallowWrapper {}`;
